File: app/src/main/java/com/tory/noname/MApplication.java
package com.tory.noname;

import com.facebook.stetho.Stetho;
import com.squareup.leakcanary.LeakCanary;
import com.squareup.leakcanary.RefWatcher;
import com.tory.library.applife.AppLifeApplication;
import com.tory.noname.main.utils.L;
import com.tory.noname.main.utils.SettingHelper;
import com.tory.noname.main.utils.Utilities;

/**
 * @Author: Tory
 * Create: 2016/9/15
 * Update: 2016/9/15
 */
public class MApplication extends AppLifeApplication {
    private static MApplication instance;
    private RefWatcher refWatcher;

    public static MApplication getInstance() {
        return instance;
    }

    public void onCreate() {
        super.onCreate();
        instance = this;
        //chrome://inspect
        Stetho.initializeWithDefaults(this);
        //if(BuildConfig.DEBUG){
        //    ReflectDebugUtil.reflectInitStetho(this);
        //}
        Utilities.setNightMode(this, SettingHelper.getInstance(this).isNightMode());
        refWatcher = setupLeakCanary();
        L.d("StethoReflection sourceDir="+getApplicationInfo().sourceDir);
    }

    private RefWatcher setupLeakCanary() {
        if (LeakCanary.isInAnalyzerProcess(this)) {
            return RefWatcher.DISABLED;
        }
        return LeakCanary.install(this);
    }
}

File: app/src/main/java/com/tory/noname/MApplication.kt
package com.tory.noname

import android.content.Context
import com.facebook.stetho.Stetho
import com.squareup.leakcanary.LeakCanary
import com.squareup.leakcanary.RefWatcher
import com.tory.library.applife.AppLifeApplication
import com.tory.library.utils.SettingHelper
import com.tory.library.utils.Utilities
import com.tory.noname.main.utils.L
import org.koin.core.context.startKoin

/**
 * @Author: Tory
 * Create: 2016/9/15
 * Update: 2016/9/15
 */
class MApplication : AppLifeApplication() {
    private var refWatcher: RefWatcher? = null

    override fun attachBaseContext(base: Context?) {
        instance = this
        super.attachBaseContext(base)
    }

    override fun onCreate() {
        super.onCreate()
        //chrome://inspect
        Stetho.initializeWithDefaults(this)
        //if(BuildConfig.DEBUG){
        //    ReflectDebugUtil.reflectInitStetho(this);
        //}
        Utilities.setNightMode(this,
            SettingHelper.getInstance(this).isNightMode)
        refWatcher = setupLeakCanary()
        L.d("StethoReflection sourceDir=" + applicationInfo.sourceDir)
        startKoin {

        }
    }

    private fun setupLeakCanary(): RefWatcher {
        return if (LeakCanary.isInAnalyzerProcess(this)) {
            RefWatcher.DISABLED
        } else LeakCanary.install(this)
    }

    companion object {
        @JvmStatic
        var instance: MApplication? = null
            private set
    }
}
